heron/executor/src/python/heron_executor.py

```diff
@@ -192,6 +201,57 @@ def __init__(self, args, shell_env):
 
     self.state_managers = []
 
+  @staticmethod
+  def parse_args(args):
+    """Uses python argparse to collect positional args"""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("shard", type=int)
+    parser.add_argument("topology_name")
+    parser.add_argument("topology_id")
+    parser.add_argument("topology_defn_file")
+    parser.add_argument("zknode")
+    parser.add_argument("zkroot")
+    parser.add_argument("tmaster_binary")
+    parser.add_argument("stmgr_binary")
+    parser.add_argument("metricsmgr_classpath")
+    parser.add_argument("instance_jvm_opts")
+    parser.add_argument("classpath")
+    parser.add_argument("master_port")
+    parser.add_argument("tmaster_controller_port")
+    parser.add_argument("tmaster_stats_port")
+    parser.add_argument("heron_internals_config_file")
+    parser.add_argument("component_rammap")
+    parser.add_argument("component_jvm_opts_in_base64")
+    parser.add_argument("pkg_type")
+    parser.add_argument("topology_bin_file")
+    parser.add_argument("heron_java_home")
+    parser.add_argument("shell_port")
+    parser.add_argument("heron_shell_binary")
+    parser.add_argument("metricsmgr_port")
+    parser.add_argument("cluster")
+    parser.add_argument("role")
+    parser.add_argument("environ")
+    parser.add_argument("instance_classpath")
+    parser.add_argument("metrics_sinks_config_file")
+    parser.add_argument("scheduler_classpath")
+    parser.add_argument("scheduler_port")
+    parser.add_argument("python_instance_binary")
+
+    parsed_args, unknown_args = parser.parse_known_args(args[1:])
+
+    if unknown_args:
+      Log.error('Unknown argument: %s' % unknown_args[0])
+      parser.print_help()
```

Comment:
In that case argparse is supposed to dump the list of required arguments: https://docs.python.org/3/library/argparse.html#usage

It's probably not the best possible help message and we should likely take some time at some point to document the individual arguments, but this will at least clue the developer in as to which argument they added doesn't belong.